sentry-spring/src/main/java/io/sentry/spring/tracing/SentrySpanClientHttpRequestInterceptor.java

```diff
@@ -27,28 +29,44 @@ public SentrySpanClientHttpRequestInterceptor(final @NotNull IHub hub) {
       @NotNull byte[] body,
       @NotNull ClientHttpRequestExecution execution)
       throws IOException {
-    final ISpan activeSpan = hub.getSpan();
-    if (activeSpan == null) {
-      return execution.execute(request, body);
-    }
+    Integer responseStatusCode = null;
+    try {
+      final ISpan activeSpan = hub.getSpan();
+      if (activeSpan == null) {
+        return execution.execute(request, body);
+      }
 
-    final ISpan span = activeSpan.startChild("http.client");
-    span.setDescription(request.getMethodValue() + " " + request.getURI());
+      final ISpan span = activeSpan.startChild("http.client");
+      span.setDescription(request.getMethodValue() + " " + request.getURI());
 
-    final SentryTraceHeader sentryTraceHeader = span.toSentryTrace();
-    request.getHeaders().add(sentryTraceHeader.getName(), sentryTraceHeader.getValue());
-    try {
-      final ClientHttpResponse response = execution.execute(request, body);
-      // handles both success and error responses
-      span.setStatus(SpanStatus.fromHttpStatusCode(response.getRawStatusCode()));
-      return response;
-    } catch (Exception e) {
-      // handles cases like connection errors
-      span.setThrowable(e);
-      span.setStatus(SpanStatus.INTERNAL_ERROR);
-      throw e;
+      final SentryTraceHeader sentryTraceHeader = span.toSentryTrace();
+      request.getHeaders().add(sentryTraceHeader.getName(), sentryTraceHeader.getValue());
+      try {
+        final ClientHttpResponse response = execution.execute(request, body);
+        // handles both success and error responses
+        span.setStatus(SpanStatus.fromHttpStatusCode(response.getRawStatusCode()));
+        responseStatusCode = response.getRawStatusCode();
+        return response;
+      } catch (Exception e) {
+        // handles cases like connection errors
+        span.setThrowable(e);
+        span.setStatus(SpanStatus.INTERNAL_ERROR);
+        throw e;
+      } finally {
+        span.finish();
+      }
     } finally {
-      span.finish();
+      addBreadcrumb(request, body, responseStatusCode);
     }
   }
+
+  private void addBreadcrumb(
+      final @NotNull HttpRequest request,
+      final @NotNull byte[] body,
+      final @Nullable Integer responseStatusCode) {
+    final Breadcrumb breadcrumb =
+        Breadcrumb.http(request.getURI().toString(), request.getMethodValue(), responseStatusCode);
+    breadcrumb.setData("requestBodySize", body.length);
```

Comment:
can we read the `response` and if available, also adding `responseBodySize`?

Reply:
`response` is an input stream here, to get the size we would need to read it.